pandas/tests/indexing/test_interval.py

```diff
@@ -122,6 +148,46 @@ def test_with_slices(self):
         with pytest.raises(ValueError):
             s[0:4:2]
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_with_slices_updated_behavior(self):
+
+        s = self.s
+
+        # slice of interval
+        expected = s.iloc[4:]
+        result = s.loc[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        expected = s.iloc[4:]
+        result = s[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='both'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='both'):]
+
+        # slice of scalar
+        with pytest.raises(NotImplementedError):
+            s[0:4] ## not sure what the behvaior should be here.
```

Comment:
Yes, slicing with integers in `[]` is always* positional, so I think should be here as well

*always = for all types of indexes expect ... ra ra .. floats!

But I am fine with raising a NotImplementedError to start with if this is not yet implemented correctly, but it *seems* this is already working on master:

```
In [33]: df = pd.DataFrame(['a', 'b', 'c'], columns=['col'], 
    ...:           index=pd.IntervalIndex.from_tuples([(11, 13), (16, 17), (14, 15)], closed='both'))
    ...: 

In [34]: df
Out[34]: 
         col
[11, 13]   a
[16, 17]   b
[14, 15]   c

In [35]: df[:2]
Out[35]: 
         col
[11, 13]   a
[16, 17]   b
```